### ISSUE TITLE
cargo chef cook panics in Docker build

### ISSUE THREAD

**Issue by benwis** [LukeMathWalker/cargo-chef#83]
Hello!
I'm working on deploying my workspace app with Docker and cargo chef, but I'm getting a panic trying to build it. Not sure what would be useful for you to help me debug it, let me know if I can provide anything.

```rust
benwis@Diziet vidette_rust % docker build --tag vidette --file Dockerfile .

[+] Building 126.8s (16/22)                                                                                            
 => [internal] load build definition from Dockerfile                                                              0.0s
 => => transferring dockerfile: 1.23kB                                                                            0.0s
 => [internal] load .dockerignore                                                                                 0.0s
 => => transferring context: 2B                                                                                   0.0s
 => [internal] load metadata for docker.io/library/debian:buster-slim                                             1.6s
 => [internal] load metadata for docker.io/library/rust:1.53.0                                                    1.6s
 => [internal] load metadata for docker.io/lukemathwalker/cargo-chef:latest-rust-1.53.0                           1.6s
 => [planner 1/4] FROM docker.io/lukemathwalker/cargo-chef:latest-rust-1.53.0@sha256:112f5524c4fb257ce9190ed8e74  0.0s
 => [internal] load build context                                                                                 0.3s
 => => transferring context: 551.51kB                                                                             0.3s
 => [builder 1/6] FROM docker.io/library/rust:1.53.0@sha256:54b76dd38a95fae4cdd57ce9742cdee01f4cb1ac6b9184b7d2a6  0.0s
 => [runtime 1/4] FROM docker.io/library/debian:buster-slim@sha256:1b138699146ca36569f2f2098c8e22c56756b5776f766  0.0s
 => CACHED [runtime 2/4] WORKDIR /app                                                                             0.0s
 => CACHED [builder 2/6] WORKDIR /app                                                                             0.0s
 => CACHED [planner 2/4] WORKDIR /app                                                                             0.0s
 => [planner 3/4] COPY . .                                                                                        3.0s
 => [planner 4/4] RUN cargo chef prepare  --recipe-path recipe.json                                               0.8s
 => CACHED [cacher 3/4] COPY --from=planner /app/recipe.json recipe.json                                          0.0s
 => ERROR [cacher 4/4] RUN RUST_BACKTRACE=1 cargo chef cook --release --recipe-path recipe.json                 121.0s
------
 > [cacher 4/4] RUN RUST_BACKTRACE=1 cargo chef cook --release --recipe-path recipe.json:
#16 2.378     Updating crates.io index
#16 94.21     Updating git repository `https://github.com/async-graphql/async-graphql/`
#16 97.76     Updating git submodule `https://github.com/async-graphql/examples`
#16 98.67     Updating git repository `https://github.com/Keats/jsonwebtoken`
#16 99.71     Updating git repository `https://github.com/launchbadge/sqlx.git`
#16 104.9  Downloading crates ...
#16 106.6   Downloaded actix-utils v3.0.0
#16 106.7   Downloaded block-buffer v0.7.3
#16 106.7   Downloaded bigdecimal v0.2.0
#16 106.7   Downloaded anyhow v1.0.42
#16 106.7   Downloaded byte-tools v0.3.1
#16 106.8   Downloaded bytestring v1.0.0
#16 106.8   Downloaded const_fn v0.4.8
#16 106.8   Downloaded chrono v0.4.19
#16 106.9   Downloaded darling_macro v0.12.4
#16 106.9   Downloaded convert_case v0.4.0
#16 106.9   Downloaded crypto-mac v0.10.1
#16 106.9   Downloaded foreign-types-shared v0.1.1
#16 106.9   Downloaded futures-channel v0.3.16
#16 106.9   Downloaded futures-sink v0.3.16
#16 106.9   Downloaded http-body v0.4.3
#16 106.9   Downloaded ident_case v1.0.1
#16 107.0   Downloaded hyper-rustls v0.22.1
#16 107.0   Downloaded subtle v2.4.1
#16 107.0   Downloaded time v0.1.44
#16 107.0   Downloaded instant v0.1.10
#16 107.0   Downloaded tap v1.0.1
#16 107.0   Downloaded httparse v1.4.1
#16 107.0   Downloaded getrandom v0.2.3
#16 107.0   Downloaded log v0.4.14
#16 107.0   Downloaded miniz_oxide v0.4.4
#16 107.1   Downloaded hyper v0.14.11
#16 107.1   Downloaded fnv v1.0.7
#16 107.1   Downloaded h2 v0.3.3
#16 107.1   Downloaded standback v0.2.17
#16 107.2   Downloaded thiserror v1.0.26
#16 107.2   Downloaded crc32fast v1.2.1
#16 107.2   Downloaded foreign-types v0.3.2
#16 107.2   Downloaded url v2.2.2
#16 107.2   Downloaded lazy_static v1.4.0
#16 107.2   Downloaded jobserver v0.1.23
#16 107.2   Downloaded uuid v0.8.2
#16 107.3   Downloaded idna v0.2.3
#16 107.3   Downloaded time-macros-impl v0.1.2
#16 107.3   Downloaded time-macros v0.1.1
#16 107.3   Downloaded unicode-normalization v0.1.19
#16 107.3   Downloaded language-tags v0.3.2
#16 107.3   Downloaded futures-intrusive v0.4.0
#16 107.4   Downloaded hex v0.4.3
#16 107.4   Downloaded dotenv_codegen v0.15.0
#16 107.4   Downloaded dotenv_codegen_implementation v0.15.0
#16 107.4   Downloaded byteorder v1.4.3
#16 107.4   Downloaded bitflags v1.2.1
#16 107.4   Downloaded ahash v0.7.4
#16 107.4   Downloaded strum v0.21.0
#16 107.4   Downloaded remove_dir_all v0.5.3
#16 107.4   Downloaded reqwest v0.11.4
#16 107.5   Downloaded tinyvec_macros v0.1.0
#16 107.5   Downloaded strum_macros v0.21.1
#16 107.5   Downloaded regex v1.5.4
#16 107.5   Downloaded socket2 v0.4.1
#16 107.6   Downloaded spin v0.9.2
#16 107.6   Downloaded tokio-rustls v0.22.0
#16 107.6   Downloaded lexical-core v0.7.6
#16 107.6   Downloaded http v0.2.4
#16 107.7   Downloaded try-lock v0.2.3
#16 107.7   Downloaded rand v0.8.4
#16 107.7   Downloaded semver-parser v0.10.2
#16 107.7   Downloaded serde_derive v1.0.127
#16 107.7   Downloaded serde v0.8.23
#16 107.7   Downloaded semver v0.11.0
#16 107.7   Downloaded typenum v1.13.0
#16 107.7   Downloaded scopeguard v1.1.0
#16 107.8   Downloaded quote v1.0.9
#16 107.8   Downloaded serde_test v0.8.23
#16 107.8   Downloaded serde-aux v2.2.0
#16 107.8   Downloaded static_assertions v1.1.0
#16 107.8   Downloaded rust-ini v0.13.0
#16 107.8   Downloaded serde_urlencoded v0.7.0
#16 107.8   Downloaded signal-hook-registry v1.4.0
#16 107.8   Downloaded ryu v1.0.5
#16 107.8   Downloaded slab v0.4.4
#16 107.8   Downloaded sha2 v0.9.5
#16 107.8   Downloaded simple_asn1 v0.5.4
#16 107.8   Downloaded tokio-native-tls v0.3.0
#16 107.9   Downloaded spin v0.5.2
#16 107.9   Downloaded rustc_version v0.3.3
#16 107.9   Downloaded regex-syntax v0.6.25
#16 107.9   Downloaded toml v0.5.8
#16 107.9   Downloaded encoding_rs v0.8.28
#16 108.1   Downloaded tracing v0.1.26
#16 108.1   Downloaded twoway v0.2.2
#16 108.1   Downloaded tokio-stream v0.1.7
#16 108.2   Downloaded tokio-macros v1.3.0
#16 108.2   Downloaded serde_json v1.0.66
#16 108.2   Downloaded rustls v0.19.1
#16 108.2   Downloaded sha-1 v0.8.2
#16 108.2   Downloaded radium v0.5.3
#16 108.2   Downloaded serde v1.0.127
#16 108.2   Downloaded proc-macro2 v1.0.28
#16 108.3   Downloaded sct v0.6.1
#16 108.3   Downloaded pin-utils v0.1.0
#16 108.3   Downloaded proc-macro-crate v1.0.0
#16 108.3   Downloaded pin-project-lite v0.2.7
#16 108.3   Downloaded pin-project-internal v1.0.8
#16 108.3   Downloaded serde-hjson v0.9.1
#16 108.3   Downloaded pin-project v1.0.8
#16 108.3   Downloaded tracing-core v0.1.18
#16 108.3   Downloaded rand_chacha v0.3.1
#16 108.3   Downloaded proc-macro-nested v0.1.7
#16 108.4   Downloaded pest_meta v2.1.3
#16 108.4   Downloaded pest_generator v2.1.3
#16 108.4   Downloaded smallvec v1.6.1
#16 108.4   Downloaded indexmap v1.7.0
#16 108.4   Downloaded sha-1 v0.9.7
#16 108.4   Downloaded stringprep v0.1.2
#16 108.4   Downloaded rand_core v0.6.3
#16 108.4   Downloaded ppv-lite86 v0.2.10
#16 108.4   Downloaded pkg-config v0.3.19
#16 108.4   Downloaded actix_derive v0.6.0
#16 108.4   Downloaded actix-web-httpauth v0.6.0-beta.2
#16 108.4   Downloaded actix-web v4.0.0-beta.8
#16 108.5   Downloaded tokio-util v0.6.7
#16 108.5   Downloaded actix-web-actors v4.0.0-beta.6
#16 108.5   Downloaded crc v2.0.0
#16 108.5   Downloaded generic-array v0.14.4
#16 108.5   Downloaded actix-web-codegen v0.5.0-beta.3
#16 108.5   Downloaded futures-util v0.3.16
#16 108.6   Downloaded futures-io v0.3.16
#16 108.6   Downloaded funty v1.1.0
#16 108.6   Downloaded adler v1.0.2
#16 108.6   Downloaded futures-macro v0.3.16
#16 108.6   Downloaded generic-array v0.12.4
#16 108.6   Downloaded ucd-trie v0.1.3
#16 108.6   Downloaded tower-service v0.3.1
#16 108.6   Downloaded time v0.2.27
#16 108.6   Downloaded claim v0.5.0
#16 108.6   Downloaded crc-catalog v1.1.1
#16 108.6   Downloaded block-buffer v0.9.0
#16 108.6   Downloaded syn v1.0.74
#16 108.7   Downloaded dirs v3.0.2
#16 108.7   Downloaded num_cpus v1.13.0
#16 108.7   Downloaded hashbrown v0.11.2
#16 108.7   Downloaded futures-task v0.3.16
#16 108.7   Downloaded unicode-segmentation v1.8.0
#16 108.7   Downloaded unicode-xid v0.2.2
#16 108.7   Downloaded futures-core v0.3.16
#16 108.8   Downloaded fake-simd v0.1.2
#16 108.8   Downloaded crossbeam-channel v0.5.1
#16 108.8   Downloaded cpufeatures v0.1.5
#16 108.8   Downloaded futures v0.3.16
#16 108.8   Downloaded flate2 v1.0.20
#16 108.8   Downloaded darling_core v0.12.4
#16 108.8   Downloaded itoa v0.4.7
#16 108.8   Downloaded hyper-tls v0.5.0
#16 108.9   Downloaded futures-executor v0.3.16
#16 108.9   Downloaded httpdate v1.0.1
#16 108.9   Downloaded heck v0.3.3
#16 108.9   Downloaded dirs-sys v0.3.6
#16 108.9   Downloaded darling v0.12.4
#16 108.9   Downloaded tempfile v3.2.0
#16 108.9   Downloaded sqlformat v0.1.6
#16 108.9   Downloaded opaque-debug v0.3.0
#16 108.9   Downloaded strsim v0.10.0
#16 108.9   Downloaded maplit v1.0.2
#16 108.9   Downloaded hmac v0.10.1
#16 108.9   Downloaded hashlink v0.7.0
#16 108.9   Downloaded futures-timer v3.0.2
#16 108.9   Downloaded derive_more v0.99.16
#16 109.0   Downloaded unicode_categories v0.1.1
#16 109.0   Downloaded version_check v0.9.3
#16 109.0   Downloaded ring v0.16.20
#16 109.5   Downloaded webpki-roots v0.21.1
#16 109.5   Downloaded tinyvec v1.3.1
#16 109.6   Downloaded whoami v1.1.2
#16 109.6   Downloaded unicode-bidi v0.3.5
#16 109.6   Downloaded aho-corasick v0.7.18
#16 109.6   Downloaded actix-codec v0.4.0
#16 109.6   Downloaded autocfg v1.0.1
#16 109.6   Downloaded ipnet v2.3.1
#16 109.6   Downloaded want v0.3.0
#16 109.6   Downloaded wyz v0.2.0
#16 109.6   Downloaded yaml-rust v0.4.5
#16 109.6   Downloaded async-trait v0.1.51
#16 109.7   Downloaded untrusted v0.7.1
#16 109.7   Downloaded arrayvec v0.5.2
#16 109.7   Downloaded async-stream v0.3.2
#16 109.7   Downloaded unchecked-index v0.2.2
#16 109.7   Downloaded webpki v0.21.4
#16 109.7   Downloaded base64ct v1.0.0
#16 109.7   Downloaded argon2 v0.2.2
#16 109.7   Downloaded actix v0.12.0
#16 109.7   Downloaded Inflector v0.11.4
#16 109.7   Downloaded thiserror-impl v1.0.26
#16 109.8   Downloaded crossbeam-queue v0.3.2
#16 109.8   Downloaded cc v1.0.69
#16 109.8   Downloaded form_urlencoded v1.0.1
#16 109.8   Downloaded cookie v0.15.1
#16 109.8   Downloaded either v1.6.1
#16 109.8   Downloaded dotenv v0.15.0
#16 109.8   Downloaded digest v0.9.0
#16 109.8   Downloaded digest v0.8.1
#16 109.8   Downloaded crypto-mac v0.8.0
#16 109.8   Downloaded matches v0.1.8
#16 109.8   Downloaded async-stream-impl v0.3.2
#16 109.8   Downloaded crossbeam-utils v0.8.5
#16 109.8   Downloaded bytes v1.0.1
#16 109.8   Downloaded cfg-if v1.0.0
#16 109.8   Downloaded config v0.10.1
#16 109.9   Downloaded brotli-sys v0.3.2
#16 109.9   Downloaded bitvec v0.19.5
#16 109.9   Downloaded atoi v0.4.0
#16 109.9   Downloaded brotli2 v0.3.2
#16 110.0   Downloaded block-padding v0.1.5
#16 110.0   Downloaded blake2 v0.9.1
#16 110.0   Downloaded actix-router v0.2.7
#16 110.0   Downloaded actix-rt v2.2.0
#16 110.0   Downloaded actix-tls v3.0.0-beta.5
#16 110.0   Downloaded md-5 v0.9.1
#16 110.0   Downloaded mime v0.3.16
#16 110.0   Downloaded local-waker v0.1.1
#16 110.0   Downloaded linked-hash-map v0.5.4
#16 110.0   Downloaded local-channel v0.1.2
#16 110.0   Downloaded linked-hash-map v0.3.0
#16 110.0   Downloaded actix-server v2.0.0-beta.5
#16 110.0   Downloaded base64 v0.13.0
#16 110.0   Downloaded native-tls v0.2.8
#16 110.0   Downloaded lru v0.6.6
#16 110.1   Downloaded actix-service v2.0.0
#16 110.1   Downloaded nom v5.1.2
#16 110.1   Downloaded multer v2.0.1
#16 110.1   Downloaded openssl-probe v0.1.4
#16 110.1   Downloaded once_cell v1.8.0
#16 110.1   Downloaded opaque-debug v0.2.3
#16 110.2   Downloaded num-traits v0.1.43
#16 110.2   Downloaded memchr v2.4.0
#16 110.2   Downloaded pem v0.8.3
#16 110.2   Downloaded password-hash v0.2.2
#16 110.2   Downloaded num-traits v0.2.14
#16 110.2   Downloaded percent-encoding v2.1.0
#16 110.2   Downloaded paste v1.0.5
#16 110.2   Downloaded parking_lot_core v0.8.3
#16 110.2   Downloaded parking_lot v0.11.1
#16 110.3   Downloaded num-integer v0.1.44
#16 110.3   Downloaded num-bigint v0.4.0
#16 110.3   Downloaded nom v6.1.2
#16 110.3   Downloaded pest_derive v2.1.0
#16 110.3   Downloaded validator_types v0.14.0
#16 110.3   Downloaded openssl-sys v0.9.65
#16 110.3   Downloaded pest v2.1.3
#16 110.4   Downloaded mio v0.7.13
#16 110.4   Downloaded tokio v1.9.0
#16 110.5   Downloaded lock_api v0.4.4
#16 110.5   Downloaded zstd-safe v3.1.0+zstd.1.4.9
#16 110.5   Downloaded openssl v0.10.35
#16 110.6   Downloaded zstd v0.7.0+zstd.1.4.9
#16 110.6   Downloaded validator v0.14.0
#16 110.6   Downloaded proc-macro-hack v0.5.19
#16 110.6   Downloaded actix-macros v0.2.1
#16 110.6   Downloaded actix-http v3.0.0-beta.9
#16 110.6   Downloaded num-bigint v0.3.2
#16 110.7   Downloaded libc v0.2.99
#16 110.8   Downloaded zstd-sys v1.5.0+zstd.1.4.9
#16 111.7  Downloading crates ...
#16 111.9   Downloaded jsonpath_lib v0.3.0
#16 111.9  Downloading crates ...
#16 112.1   Downloaded wiremock v0.5.6
#16 112.5    Compiling libc v0.2.99
#16 112.5    Compiling autocfg v1.0.1
#16 112.5    Compiling proc-macro2 v1.0.28
#16 112.5    Compiling syn v1.0.74
#16 120.5 thread 'main' panicked at 'Process terminated by signal', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/cargo-chef-0.1.21/src/recipe.rs:137:21
#16 120.6 stack backtrace:
#16 120.9    0: std::panicking::begin_panic
#16 120.9    1: chef::recipe::Recipe::cook
#16 120.9    2: cargo_chef::_main
#16 120.9 note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
------
executor failed running [/bin/sh -c RUST_BACKTRACE=1 cargo chef cook --release --recipe-path recipe.json]: exit code: 101
 ```
Cargo.toml for main crate
```rust
[package]
name = "vidette"
version = "0.1.0"
edition = "2018"

[lib]
path = "src/lib.rs"

[[bin]]
path = "src/main.rs"
name = "vidette"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
email = {path="../email"}
async-graphql = { git = "https://github.com/async-graphql/async-graphql/", branch="actix-web-v4-beta", features = ["uuid", "chrono", "dataloader"] }
slab = "0.4.2"
uuid = { version = "0.8", features = ["serde", "v4"] }
chrono = {version= "0.4", features=["serde"]}
serde_derive = "1.0"
bigdecimal = {version="0.2", features=["serde"]}
async-trait = "0.1.50"
strum = "0.21.0"
strum_macros = "0.21.1"
anyhow = "1.0"
argon2 = "0.2"
rand_core = { version = "0.6", features = ["std"] }
thiserror = "1.0"
jsonwebtoken = { git = "https://github.com/Keats/jsonwebtoken", branch = "next" }
regex = "1.5"
futures = "0.3"
actix-web = { version = "4.0.0-beta.8" }
dotenv = "0.15.0"
dotenv_codegen = "0.15.0"
time = { version = "0.2", features = ["serde"] }
http = "0.2"
futures-util = "0.3.15"
actix-web-httpauth = "0.6.0-beta.2"
async-graphql-actix-web = { git = "https://github.com/async-graphql/async-graphql/", branch="actix-web-v4-beta" }
config = { version = "0.10.1", features = ["yaml"] }
serde = { version = "1.0.126", features = ["derive"] }
serde_json = "1.0.64"
sqlx = { git = "https://github.com/launchbadge/sqlx.git", features = [ "runtime-actix-native-tls", "macros","bigdecimal", "postgres", "uuid", "json", "chrono", "migrate", "offline"] }
serde-aux = "2.2.0"

[dev-dependencies]
jsonpath_lib = "0.3.0"
wiremock = {version="0.5"}
uuid = { version = "0.8", features = ["serde", "v4"] }
reqwest = { version = "0.11", features = ["json", "rustls-tls"] }
tokio = { version = "1", features = ["full"] }

```
Cargo.toml for tests crate
```rust
[package]
name = "vidette_tests"
version = "0.1.0"
edition = "2018"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
actix-rt = "2"
vidette = {path="../vidette"}

[dev-dependencies]
jsonpath_lib = "0.3.0"
wiremock = {version="0.5"}
uuid = { version = "0.8", features = ["serde", "v4"] }
tokio = { version = "1", features = ["full"] }
# sqlx = { version = "0.5", features = [ "runtime-actix-native-tls", "macros","bigdecimal", "postgres", "uuid", "json", "chrono", "migrate", "offline"] }
sqlx = { git = "https://github.com/launchbadge/sqlx.git", features = [ "runtime-actix-native-tls", "macros","bigdecimal", "postgres", "uuid", "json", "chrono", "migrate", "offline"] }

futures = "0.3"
serde = { version = "1.0.126", features = ["derive"] }
serde_json = "1.0.64"
actix-web = { version = "4.0.0-beta.8" }
anyhow = "1.0"
reqwest = { version = "0.11",features = ["json", "rustls-tls"] }

```
Thank you for any help you can provide!

**Comment by LukeMathWalker** [comment]
The panic is coming from the `cargo build` command launched by `cargo-chef`.
Is the source code of the application public?

**Comment by benwis** [comment]
Not really. The odd part is that I can run cargo build --release outside of docker and it works or I can create a Dockerfile without cargo chef it and it works as well. Let me put together something to look at for you :)

**Comment by chusteven** [comment]
Hello sorry to add to this; but I'm running into the same issue 😅 ; I too can run `cargo build --release` outside of Docker (on M1 2020 MacBook Pro); below is the output from running the same inside Docker.

For whatever it's worth I notice that twice in a row now it has failed at the step `Compiling proc-macro2 v1.0.32`
```
@ zero2prod [stevenchu]$ docker build --tag zero2prod --file Dockerfile .
[+] Building 110.5s (14/20)
 => [internal] load build definition from Dockerfile                                                                                                                                            0.0s
 => => transferring dockerfile: 795B                                                                                                                                                            0.0s
 => [internal] load .dockerignore                                                                                                                                                               0.0s
 => => transferring context: 34B                                                                                                                                                                0.0s
 => [internal] load metadata for docker.io/library/debian:bullseye-slim                                                                                                                         1.2s
 => [internal] load metadata for docker.io/lukemathwalker/cargo-chef:latest-rust-1.56.0                                                                                                         1.3s
 => [auth] lukemathwalker/cargo-chef:pull token for registry-1.docker.io                                                                                                                        0.0s
 => [auth] library/debian:pull token for registry-1.docker.io                                                                                                                                   0.0s
 => [runtime 1/5] FROM docker.io/library/debian:bullseye-slim@sha256:a23887a2e830b815955e010f30d4c2430cd5ef82e93c130471024bc9f808d5d3                                                           0.0s
 => [internal] load build context                                                                                                                                                               0.1s
 => => transferring context: 20.88kB                                                                                                                                                            0.0s
 => [chef 1/2] FROM docker.io/lukemathwalker/cargo-chef:latest-rust-1.56.0@sha256:94c1c11d7afcf4c26fbeec11bbcdc42079bb7fcf536a92f4e769472373fe8e95                                              0.0s
 => CACHED [chef 2/2] WORKDIR /app                                                                                                                                                              0.0s
 => CACHED [planner 1/2] COPY . .                                                                                                                                                               0.0s
 => CACHED [planner 2/2] RUN cargo chef prepare --recipe-path recipe.json                                                                                                                       0.0s
 => CACHED [builder 1/4] COPY --from=planner /app/recipe.json recipe.json                                                                                                                       0.0s
 => ERROR [builder 2/4] RUN RUST_BACKTRACE=full cargo chef cook --release --recipe-path recipe.json                                                                                           109.1s
------
 > [builder 2/4] RUN RUST_BACKTRACE=full cargo chef cook --release --recipe-path recipe.json:
#16 2.908     Updating crates.io index
#16 83.51  Downloading crates ...
#16 90.15   Downloaded arrayvec v0.5.2
#16 90.31   Downloaded cpufeatures v0.2.1
#16 90.32   Downloaded crc32fast v1.2.1
...
#16 95.21    Compiling proc-macro2 v1.0.32
#16 108.6 thread 'main' panicked at 'Process terminated by signal', /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/cargo-chef-0.1.31/src/recipe.rs:146:21
#16 108.7 stack backtrace:
#16 108.8    0:       0x4000268afc - std::backtrace_rs::backtrace::libunwind::trace::hd3a6cb8e32f9fc43
#16 108.8                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/../../backtrace/src/backtrace/libunwind.rs:90:5
#16 108.8    1:       0x4000268afc - std::backtrace_rs::backtrace::trace_unsynchronized::h6dcceebcd6a9f2ab
#16 108.8                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
#16 108.8    2:       0x4000268afc - std::sys_common::backtrace::_print_fmt::h4eacf4e9ad977499
#16 108.8                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/sys_common/backtrace.rs:67:5
#16 108.8    3:       0x4000268afc - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h1e42c22572a7e7b4
#16 108.8                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/sys_common/backtrace.rs:46:22
#16 108.8    4:       0x400028d96c - core::fmt::write::h072b578a8e90aa54
#16 108.8                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/fmt/mod.rs:1150:17
#16 108.9    5:       0x40002659b5 - std::io::Write::write_fmt::hc32d3cabefff1d24
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/io/mod.rs:1667:15
#16 108.9    6:       0x400026a9d0 - std::sys_common::backtrace::_print::h058b85b6efc15780
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/sys_common/backtrace.rs:49:5
#16 108.9    7:       0x400026a9d0 - std::sys_common::backtrace::print::hdb7de10b704104b4
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/sys_common/backtrace.rs:36:9
#16 108.9    8:       0x400026a9d0 - std::panicking::default_hook::{{closure}}::h5451997d203bda2f
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panicking.rs:210:50
#16 108.9    9:       0x400026a587 - std::panicking::default_hook::h419b121e68f746a2
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panicking.rs:227:9
#16 108.9   10:       0x400026b084 - std::panicking::rust_panic_with_hook::h32c44b1364c1c247
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panicking.rs:624:17
#16 108.9   11:       0x40000bdb55 - std::panicking::begin_panic::{{closure}}::h68dc90ee5ef28ac3
#16 108.9   12:       0x40000bdae6 - std::sys_common::backtrace::__rust_end_short_backtrace::h85efabf3c69d7f31
#16 108.9   13:       0x400003ee21 - std::panicking::begin_panic::hfb38d3ffa2926355
#16 108.9   14:       0x40000bf8f8 - chef::recipe::Recipe::cook::hcca3a1b1f5ee4e17
#16 108.9   15:       0x4000056d31 - cargo_chef::_main::hf6bdbe67bee919d1
#16 108.9   16:       0x4000050fe3 - std::sys_common::backtrace::__rust_begin_short_backtrace::h00bf14c24ead7463
#16 108.9   17:       0x400005af9d - std::rt::lang_start::{{closure}}::hc2ce9ee509d0a1ac
#16 108.9   18:       0x400026b58a - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::h97e9d5f21cbee92f
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/core/src/ops/function.rs:259:13
#16 108.9   19:       0x400026b58a - std::panicking::try::do_call::hdfecd950eba8f698
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panicking.rs:403:40
#16 108.9   20:       0x400026b58a - std::panicking::try::h3c77c8ba3eb0735c
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panicking.rs:367:19
#16 108.9   21:       0x400026b58a - std::panic::catch_unwind::h8e0d62076d8d837e
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panic.rs:129:14
#16 108.9   22:       0x400026b58a - std::rt::lang_start_internal::{{closure}}::h647045b4c6a22e67
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/rt.rs:45:48
#16 108.9   23:       0x400026b58a - std::panicking::try::do_call::h905d96e4a757647e
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panicking.rs:403:40
#16 108.9   24:       0x400026b58a - std::panicking::try::hc113426beb3c908b
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panicking.rs:367:19
#16 108.9   25:       0x400026b58a - std::panic::catch_unwind::h74f684ea7038ea12
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/panic.rs:129:14
#16 108.9   26:       0x400026b58a - std::rt::lang_start_internal::h5ba185919d396c97
#16 108.9                                at /rustc/09c42c45858d5f3aedfa670698275303a3d19afa/library/std/src/rt.rs:45:20
#16 108.9   27:       0x40000577c2 - main
#16 108.9   28:       0x4001bd2d0a - __libc_start_main
#16 108.9   29:       0x400004aafa - _start
#16 108.9   30:                0x0 - <unknown>
------
executor failed running [/bin/sh -c RUST_BACKTRACE=full cargo chef cook --release --recipe-path recipe.json]: exit code: 101
```

And my `Cargo.toml` file
```
@ zero2prod [stevenchu]$ cat Cargo.toml
[package]
name = "zero2prod"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[lib]
path = "src/lib.rs"

[[bin]]
path = "src/main.rs"
name = "zero2prod"

[dependencies]
actix-http = "3.0.0-beta.10"
actix-web = "4.0.0-beta.9"
chrono = "0.4.15"
config = "0.11.0"
log = "0.4"
serde = { version = "1", features = ["derive"]}
tracing = { version = "0.1", features = ["log"] }
tracing-subscriber = { version = "0.2.12", features = ["registry", "env-filter"] }
uuid = { version = "0.8.1", features = ["v4"] }
tracing-bunyan-formatter = "0.1.6"
tracing-log = "0.1.2"
tracing-futures = "0.2.5"
tracing-actix-web = "0.4.0-beta.12"
serde-aux = "3.0.1"

[dependencies.sqlx]
version = "0.5.7"
default-features = false
features = [
    "runtime-actix-rustls",
    "macros",
    "postgres",
    "uuid",
    "chrono",
    "migrate",
    "offline"
]

[dev-dependencies]
actix-rt = "2"
once_cell = "1.8.0"
reqwest = "0.11"
tokio = "1"
```

**Comment by LukeMathWalker** [comment]
It might be related to https://github.com/LukeMathWalker/cargo-chef/issues/95#issuecomment-972057146?

**Comment by chusteven** [comment]
> It might be related to [#95 (comment)](https://github.com/LukeMathWalker/cargo-chef/issues/95#issuecomment-972057146)?

oh nice! yeah sounds like it could be the culprit for at least me (since i'm running on the M1 chip); i'll give the `theooiry/cargo-chef` image a whirl :)

**Comment by LukeMathWalker** [comment]
#105 has been merged - can you confirm if the image for `cargo-chef` 0.1.33 solves your issue? @chusteven

**Comment by chusteven** [comment]
> #105 has been merged - can you confirm if the image for `cargo-chef` 0.1.33 solves your issue? @chusteven

@LukeMathWalker thank you so much! I'm not super familiar with the intricacies of Docker but I suspect there's no image yet available for Rust 1.56? Here's what I see when I search for the tag "rust-1.56.0" : https://hub.docker.com/r/lukemathwalker/cargo-chef/tags?page=1&name=rust-1.56.0

Note that there's no "linux/arm64" to be seen anywhere in the search results 😅

**Comment by LukeMathWalker** [comment]
Search for `rust-1.56.1` 😄
